pyscf/adc/uadc.py

```diff
@@ -3870,26 +3903,531 @@ def get_trans_moments(adc):
             T_bb = adc.compute_trans_moments(orb, spin = "beta")
             T_b.append(T_bb)
     
+    cput0 = log.timer_debug1("completed spec vactor calc in ADC(3) calculation", *cput0)
     return (T_a, T_b)
 
    
-def get_spec_factors(adc, T, U, nroots=1):
+def dyson_orb(adc, X_a, X_b, nroots=1):
+
+    dyson_mo_a = np.dot(adc.mo_coeff[0],x_a).reshape(-1,nroots)
+    dyson_mo_b = np.dot(adc.mo_coeff[1],x_b).reshape(-1,nroots)
+
+    return dyson_mo_a,dyson_mo_b
```

Comment:
Return as a tuple. Check other UADC quantities if they need to be returned as tuples.

Reply:
Done